docs/studios/unity/unity-sdk/creating-bots/javascript/agent-builder.mdx

```diff
@@ -0,0 +1,240 @@
+:::
+
+## What Is a Behavior Tree?
+
+<BehaviorTree />
+
+## Creating a Bot
+
+Create a new AI from the Bot Manager and select `Agent Builder` from the list of creation options
+(see this section on [Cloning a Starter Template](../../../../../players/creating-bots/starter-templates) to complete this step).
+
+![Create new AI](../../../../../common/bot-creation/img/create-bot-button.png)
+![Select Creation Option](../../../../../common/agent-builder/img/create-from-unity-template.png)
+
+## Loading Your Tree
+
+<LoadingYourTree />
+
+## Getting Started with the Agent Builder
+
+### Agent Builder and Your Game Integration
+
+At its core, our Unity SDK revolves around defining actions that your bot can perform and defining
+information about your game-state that the bot can analyze to make decisions. Actions and game-state information will be different
+for every game, so Agent Builder requires some simple setup for your unique integration.
+
+:::info
+The Agent Builder is designed for the creation of [PlayTest bots](./playtest-bots), which form decisions every tick.
+This means your Agent Builder tree will be executed once per game tick.
+
+Performing actions through our SDK queues them to be handled within your game asynchronously.
+This means a node in your tree will **not** wait for an action to complete before executing its next line of code.
+To avoid interrupting previously-queued actions, it's a good idea to include flags in your state to indicate whether the bot is ready for new instructions.
+:::
+
+
+#### Bot Configuration
+
+Click the "Bot Settings" button in the Agent Builder interface to access a range of important settings for your bot.
+See the [Bot Configuration section](./configuration) for information on each setting and how it affects bot startup and teardown.
+Bot configurations are applied at runtime before the first game-tick your tree is executed.
+
+![Bot Settings](../../../../../common/agent-builder/img/bot-settings-button.png)
+![Bot Configuration](../../../../../common/agent-builder/img/bot-configuration-tab.png)
+
+#### Saving Your Unity Context
+
+From the Unity Editor, expand the "Regression Games" menu and select "Agent Builder > Extract Data".
+This will save your Unity Context as a .zip to your top-level project directory.
+
+![Extract Game Context](../../../../../common/agent-builder/img/extract-game-context.png)
+
+In the Agent Builder, click the "Bot Settings" button, then select the .zip from the "Upload Generated Game Context" tab.
+
+![Upload Game Context](../../../../../common/agent-builder/img/upload-game-context.png)
+
+### Adding Nodes
+
+<AddingNodes />
+
+### Generating Code
+
+:::note
+All samples in this section are written in Typescript, which is the only language currently supported by the Agent Builder.
+If you'd like to see support for other languages, like C#, let us know [here](https://regression-games.sleekplan.app/feedback).
```

Comment:
To my point above, we don't know if/when C# support will land. And while we certainly should update the docs when it does, it's easiest to just describe the situation now rather than predicting the future. We are, and always will be, "constantly evaluating support for other languages".

```suggestion
All samples in this section are written in TypeScript.
We're constantly evaluating support for other languages. Feel free to request others on our [Feedback Site](https://regression-games.sleekplan.app/feedback)!.
```

Reply:
Updated! I believe Sleekplan has a public roadmap feature. I don't think we're currently using it, but it could be really helpful for simplifying our documentation. ex. "See our roadmap for language support" (@vontell)

Reply:
That's a great way to do this. Centralizing where we talk about the future makes it easier to ensure we keep it relatively up-to-date :). My main concern is having lots of different docs referencing future features that may or may not actually exist at the time the reader is looking at the page.